test/extended/machines/cluster.go

```diff
@@ -120,83 +122,169 @@ var _ = g.Describe("[sig-node] Managed cluster", func() {
 		o.Expect(err).NotTo(o.HaveOccurred())
 		o.Expect(nodes.Items).NotTo(o.HaveLen(0))
 
-		expected := make(map[string]int)
-		actual := make(map[string]int)
+		bootTimelinesByNode := make(map[string][]bootTimelineEntry)
 		errs := make([]error, 0)
 
+		allNodeLogs := &bytes.Buffer{}
 		// List nodes, set actual and expected number of reboots
 		for _, node := range nodes.Items {
-			// Examine only the nodes which are available at the start of the test
-			if !slices.Contains(staticNodeNames, node.Name) {
-				continue
-			}
-			nodeBoots, nodeReboots, err := getNumberOfBootsForNode(oc.KubeClient(), oc.Namespace(), node.Name)
+			nodeBoots, nodeReboots, nodeLogs, err := getNumberOfBootsForNode(oc.KubeClient(), oc.Namespace(), node.Name)
+			allNodeLogs.WriteString(nodeLogs + "\n\n")
 			if err != nil {
 				errs = append(errs, err)
+				continue
 			}
-			// actual - number of boots
-			// expected - number of reboots recorded by systemd-login + 1 (first boot)
-			actual[node.Name] = nodeBoots
-			expected[node.Name] = nodeReboots + 1
+			allTimelineEvents := []bootTimelineEntry{}
+			allTimelineEvents = append(allTimelineEvents, nodeBoots...)
+			allTimelineEvents = append(allTimelineEvents, nodeReboots...)
+			sort.Sort(sort.Reverse(byTime(allTimelineEvents)))
+			bootTimelinesByNode[node.Name] = allTimelineEvents
+
+			e2e.Logf("timeline events for %q\n%v", node.Name, allTimelineEvents)
 		}
-		// Use gomega's WithTransform to compare actual to expected - and check that errs is empty
-		var emptyErrors = func(a interface{}) (interface{}, error) {
-			if len(errs) > 0 {
-				return a, fmt.Errorf("errors found: %v", errs)
+		for nodeName, timelineEvents := range bootTimelinesByNode {
+			// every reboot (except maybe the first), should have a rebootRequest before it.
+			// we reversed the sort so we can step backwards in time like this
+			for i, timelineEvent := range timelineEvents {
+				// boots should be the events, reboots should be the odds
+				expectedReboot := (i % 2) == 1
+				expectedBoot := !expectedReboot
+				isActualBoot := timelineEvent.action == "Boot"
+				isActualReboot := timelineEvent.action == "RebootRequest"
```

Comment:
this is the insight I'm using for the test.  Every other entry should be a rebootrequest.  If it's not, the test should fail.